### PR TITLE
StarknetChainId replaced with ChainId

### LINKED ISSUE #297: Add support for custom chain id
**Is your feature request related to a problem? Please describe.**

starknet.py supports custom starknet instances but doesn't support custom chain id

**Describe the solution you'd like**

@Arcticae will ask SW about this issue

### PR DESCRIPTION
**Pull request by war-in**
### **Please check if the PR fulfills these requirements**
- [ ] The formatter, linter, and tests all run without an error
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Closes #297 

* **What is the current behavior?** (You can also link to an open issue here)

ChainId can be only StarknetChainId

## **What is the new behavior (if this is a feature change)?**

ChainId can be int too

## **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No

### CODE REVIEW

File: starknet_py/net/models/transaction.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def compute_invoke_hash(
         contract_address=contract_address,
         entry_point_selector=entry_point_selector,
         calldata=calldata,
-        chain_id=chain_id.value,
+        chain_id=chain_id if isinstance(chain_id, int) else chain_id.value,
```
Comment (Arcticae) [inline starknet_py/net/models/transaction.py:61]: i'd do it another way around - check if it's an enum, and only then get `.value`

File: starknet_py/net/models/transaction.py [hunk 2]
```diff
@@ -69,21 +69,21 @@ def compute_invoke_hash(
...
-    :param chain_id: StarknetChainId
+    :param chain_id: ChainId
     :param version: Version of deploy transaction
     :return: calculated hash
     """
     return calculate_deploy_transaction_hash(
         contract_address=contract_address,
         constructor_calldata=calldata,
-        chain_id=chain_id.value,
+        chain_id=chain_id if isinstance(chain_id, int) else chain_id.value,
```
Comment (Arcticae) [inline starknet_py/net/models/transaction.py:86]: same here

File: starknet_py/net/signer/stark_curve_signer.py [hunk 3]
```diff
@@ -59,7 +59,9 @@ def sign_transaction(
             entry_point_selector=transaction.entry_point_selector,
             calldata=transaction.calldata,
             max_fee=transaction.max_fee,
-            chain_id=self.chain_id.value,
+            chain_id=self.chain_id
+            if isinstance(self.chain_id, int)
+            else self.chain_id.value,
```
Comment (Arcticae) [inline starknet_py/net/signer/stark_curve_signer.py:64]: and here

**Review by Arcticae — CHANGES_REQUESTED** [review]

**Review by cptartur — COMMENTED** [review]
I'd add some test verifying if passing custom chain id works properly.

**Pull request closed without merging**